### PR TITLE
make `get_nvidia_gpu_stats` public

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

This can be a public method so that it can be used when training with Lite.
Fixes: #10414

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @tchaton @justusschock @awaelchli

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]
LGTM!

**Review by ethanwharris — COMMENTED** [review]
Looks good :smiley: Could you also update the forwarding references in https://github.com/PyTorchLightning/pytorch-lightning/blob/e61a98a3d0cbe28f1df41841b4766166dd9ea531/pytorch_lightning/utilities/memory.py#L111

**Comment by rohitgr7** [comment]
> Looks good 😃 Could you also update the forwarding references in
> 
> https://github.com/PyTorchLightning/pytorch-lightning/blob/e61a98a3d0cbe28f1df41841b4766166dd9ea531/pytorch_lightning/utilities/memory.py#L111

good catch. thanks 😃

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Comment by rohitgr7** [comment]
> @rohitgr7 I'm not sure about this. The accelerator api for accessing device stats is already public.
> 
> > This can be a public method so that it can be used when training with Lite.
> 
> Is there an issue connected to this? This to me does not seem sufficient for a motivation. In theory, everything in Lightning can become a utility. But should it?

I was just trying out Lite over the weekend. Comparing lite/lightning. So was trying to check the GPU usage on kaggle. But couldn't find a way, so thought this might be a good utility. There is no issue related to it.

File: pytorch_lightning/accelerators/gpu.py [hunk 1]
```diff
@@ -83,63 +82,3 @@ def teardown(self) -> None:
     def auto_device_count() -> int:
         """Get the devices when set to auto."""
         return torch.cuda.device_count()
-
-
-def _get_nvidia_gpu_stats(device: torch.device) -> Dict[str, float]:
```
Comment (carmocca) [inline pytorch_lightning/accelerators/gpu.py:88]: I'm not against making it public, but I would keep it in this file. the utilities file keeps growing and growing

and `from pytorch_lightning.accelerators.gpu import get_nvidia_gpu_stats` seems more natural to me.

**Comment by awaelchli** [comment]
@rohitgr7 Such a problem like you describe would have been perfect for a GitHub issue to discuss. For example, if this was one, then I would have asked what the reasons are for a user to access

- the utility in a device specific way, as opposed to a device agnostic way
- why rely on a particular call to nvidia-smi when one could simply call built-in functions https://pytorch.org/docs/stable/cuda.html#memory-management

**Comment by rohitgr7** [comment]
@awaelchli opened an issue here: https://github.com/PyTorchLightning/pytorch-lightning/issues/10414

File: pytorch_lightning/accelerators/gpu.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/accelerators/gpu.py:88]: Same here !

File: pytorch_lightning/utilities/gpu.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        ("temperature.gpu", "°C"),
+        ("temperature.memory", "°C"),
+    ]
+    gpu_stat_keys = [k for k, _ in gpu_stat_metrics]
+    gpu_query = ",".join(gpu_stat_keys)
+
+    gpu_id = _get_gpu_id(device.index)
+    nvidia_smi_path = shutil.which("nvidia-smi")
+    if nvidia_smi_path is None:
+        raise FileNotFoundError("nvidia-smi: command not found")
```
Comment (ananthsub) [inline pytorch_lightning/utilities/gpu.py:51]: this should be at the start of the function to fail faster

File: pytorch_lightning/accelerators/gpu.py (hunk 1 above)
Comment (rohitgr7) [reply pytorch_lightning/accelerators/gpu.py:88]: fine with me. I just want it to be public

Comment (rohitgr7) [reply pytorch_lightning/accelerators/gpu.py:88]: updated!

File: pytorch_lightning/utilities/gpu.py (hunk 2 above)
Comment (rohitgr7) [reply pytorch_lightning/utilities/gpu.py]: done!

**Review by carmocca — APPROVED** [review]

**Pull request merged**